addons/html_editor/static/src/others/dynamic_placeholder_plugin.js

```diff
@@ -73,14 +73,29 @@ export class DynamicPlaceholderPlugin extends Plugin {
     /**
      * @param {string} chain
      * @param {string} defaultValue
+     * @param {string} fieldType
      */
-    onValidate(chain, defaultValue) {
+    async onValidate(chain, defaultValue, fieldType) {
         if (!chain) {
             return;
         }
 
+        const { partner_id: partnerField } = await this.services.field.loadFields(this.defaultResModel);
+
+        let dynamicPlaceholder;
+        if (fieldType == 'datetime') {
```

Comment:
Don't know the internals of html editor and stuff, but seems a bit ping-pong, with an if put in the code. Invoking @abd-msyukyu-odoo as he masters that kind of code.

Reply:
@tde-banana-odoo @nihp-odoo Hello! I'm not an expert in template rendering, but I suppose there is somewhere a python function which evaluates all dynamic placeholders based on `object.${chain}`. Wouldn't it be possible to, at that time, detect that the chain points to a `Date(time)` field, and output a well formatted date based on the desired reference for the timezone?
I wouldn't do all that in the editor plugin.

If it's not possible to detect that in Python, I'd set the field type as an attribute of the `t` element, and still do the formatting in python, based on the trust that the type specified in that attribute is correct.

You shouldn't have to care about the `defaultValue` with that weird "safe/unsafe" consideration too.

Reply:
QWeb does not do that, that's the purpose of format_datetime ^^ . I suppose each side consider the other side should do it.

Reply:
Then I guess the approach in this PR is correct, we prepare a python expression which will result in the desired format, using the correct rpc calls

Reply:
Though we could create a python function which generates that python expression for us, and only call that in the plugin, with sufficient parameters

Reply:
Yep, unsure what is the best approach ^^

Reply:
Where does "chain" come from ? Maybe that part could be improved ? Strange that we consider chain is the right part, and we hardcode like "object.chain". Not really easy to have something a bit modular ^^ .